lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -67,9 +68,9 @@
     /**
      * Local ClassMetadata cache to avoid going to the EntityManager all the time.
      *
-     * @var array
+     * @var ClassMetadata[]|LazyPropertyMap indexed by class name
      */
-    protected $_metadataCache = [];
+    protected $_metadataCache;
```

Comment:
Isn't this a BC break? This class doesn't have `@internal`. Maybe we should at least add to the list.

Reply:
No BC break: the entire namespace is `Internal`

Reply:
Is there a naming convention about internal namespaces?

Reply:
Not really - we can change all these, it just isn't related to this PR